activity_calendar/models.py

```diff
@@ -312,7 +206,137 @@ def get_absolute_url(self, recurrence_id=None):
+        # Create the fields that are copied and can be tweaked from the Activity Model
+        meta_class = attrs.get('Meta')
+        if meta_class:
+            for field_name in getattr(meta_class, 'copy_fields', []):
+                # Generate the model fields
+                for field in Activity._meta.fields:
+                    if field.name == field_name:
+                        new_field = copy.deepcopy(field)
+                        new_field.blank = True
+                        new_field.null = True
+                        new_field.default = None
+
+                        new_field_name = 'local_'+field_name
+                        new_field.name = new_field_name
+
+                        attrs[new_field_name] = new_field
+                        break
+                else:
+                    raise KeyError(f"'{field_name}' is not a property on Activity")
+
+                # Generate the method that retrieves correct values where necessary
+                attrs[field_name] = property(mcs.generate_lookup_method(field_name))
+
+            # Remove the attribute from the Meta class. It's been used and Django doesn't know what to do with it
+            del meta_class.copy_fields
+
+            for field_name in getattr(meta_class, 'link_fields', []):
+                # Generate the method that retrieves correct values where necessary
+                attrs[field_name] = property(mcs.generate_lookup_method(field_name))
+
+            del meta_class.link_fields
+
+        return super().__new__(mcs, name, bases, attrs, **kwargs)
+
+    @classmethod
+    def generate_lookup_method(cls, field_name):
+        """ Generates a lookup method that for the given field_name looks in the objects local storage before looking
+        in its parent_activity model instead for the given attribute """
+        def get_activity_attribute(self):
+            local_attr = getattr(self, 'local_'+field_name, None)
+            if local_attr is None or local_attr == '':
+                return getattr(self.parent_activity, field_name)
+            return local_attr
+
+        return get_activity_attribute
+
+
+class ActivityMoment(models.Model, metaclass=ActivityDuplicate):
+    parent_activity = models.ForeignKey(Activity, on_delete=models.CASCADE)
+    recurrence_id = models.DateTimeField(verbose_name="parent activity date/time")
+
+    created_date = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        unique_together = ['parent_activity', 'recurrence_id']
+        # Define the fields that can be locally be overwritten
+        copy_fields = ['description', 'location', 'max_participants']
+        # Define fields that are instantly looked for in the parent_activity
+        # If at any point in the future these must become customisable, one only has to move the field name to the
+        # copy_fields attribute
+        link_fields = ['title', 'image']
```

Comment:
I suggest we move title already to the customisable fields. It's not uncommon to have a shared boardgame night with another association and I don't think it's needed to require it to be a separate activity.